src/App/screens/Instructor/components/InstructorLessonsCallback.js

```diff
@@ -0,0 +1,43 @@
+import {Component, PropTypes} from 'react'
+import {connect} from 'react-redux'
+import {requestInstructorLessons} from '../state/actions'
+
+export default connect(
+  null,
+  {requestInstructorLessons}
+)(class extends Component {
+  
+  static propTypes = {
+    children: PropTypes.func.isRequired,
+  }
+
+  state = {
+    currentPage: 1
+  }
+
+  fetchLessons = (currentPage = 1) => {
+    const {
+      requestInstructorLessons,
+      instructor,
+      size = 10,
+      states = undefined,
+    } = this.props
+    this.setState({currentPage})
+    requestInstructorLessons({
+      lessons_url: instructor.lessons_url,
+      page: currentPage,
+      size,
+      states,
+    })
+  }
+
+  componentWillMount() {
+    this.fetchLessons()
+  }
+
+  render() {
+    const {currentPage} = this.state
+    const {children} = this.props
+    return children(currentPage, this.fetchLessons)
+  }
+})
```

Comment:
No, the `Callback` isn't needed in the name of the file. I can remove that.

As for the implicit returns; I really like them as it removes the ease of mutation, encourages more FP / expression based rendering. I also don't use IDE stuff so I don't feel the pain of the explicit names. I like the node_modules esque setup of this where a component has an `index` and an implicit return so there aren't any hard-coded names other than the parent folder name. Makes me feel warm and fuzzy :) But if you really want me to I can keep all the explicitly named stuff.